File: clojure/src/pehape/string/implode.clj
(ns pehape.string.implode
  (:require [clojure.string :as str]))


(defn implode
  "The Implode function joins array elements with a string.

   Arguments:
   - `separator` Specifies what to put between the array elements.
   - `array` The array to join to a string.
   Returns:
   - A string from elements of an array"
  {:added "0.0.1"}
  [separator array]
  (str/join separator array))

File: rust/src/string/implode.rs
/// The Implode function joins array elements with a string.
///
/// **Arguments:**
///
/// * `separator` - Specifies what to put between the array elements.
/// * `array` - The array to join to a string.
///
/// **Returns:**
///
/// A string from elements of an array
pub fn implode<T>(separator: &str, array: &[T]) -> String
where
    T: ToString,
{
    array
        .iter()
        .map(|v| v.to_string())
        .collect::<Vec<String>>()
        .join(separator)
}
